File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        //store frequencies in unordered map
        unordered_map<int, int> freq;
        for(int num : arr) {
            freq[num]++;
        }

        //store the frequencies in unordered set
        unordered_set<int> freqSet;
        for (auto [key, value] : freq) {
            freqSet.insert(value);
        }

        //if the set size equal to map size freq are unique
        return freqSet.size() == freq.size();
        
    } //runtime : O(N), space : O(N)
};

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        return len(set(Counter(arr).values())) == len(set(arr))
        
        
        
        
#         unordered_map = {}
#         for num in arr:
#             if num in unordered_map:
#                 unordered_map[num] = unordered_map[num] + 1
#             else:
#                 unordered_map[num] = 1
                
#         return len(unordered_map.keys()) == len(set(unordered_map.values()))
        